integration-test/helpers/harness.js

```diff
@@ -63,35 +66,34 @@ const setup = async (ops) => {
 
     bgPage.on('request', (req) => { requests.push(req.url()) })
 
-    return { browser, bgPage, requests }
+    return { browser, bgPage, requests, teardown: teardown.bind(null, browser, dataDir) }
```

Comment:
Just so I understand, is this to prevent multiple tests at the same time running?

Can we use a closure instead, largely we avoid `bind` calls.

Reply:
It's mostly just to bind a teardown method for each call to `setup()` - this allows us to cleanly dispose of temporary data without having to introduce another param that would need to be given to 'teardown'.

For example, to correctly cleanup after a test run, without the introduced closure/bound function, we'd end up with:

**before**
```js
// in setup 
({browser, dataDir}) = await harness.setup();

// later in cleanup
await harness.teardown(browser, dataDir)
```

Instead, I suggest we switch to the well-used pattern of the `setup()` call returning it's own teardown logic. This gives us scope in the future to increase/decrease the work done in teardown without it affecting the interface.

**after**
```js
// in setup 
({browser, teardown}) = await harness.setup();

// later in cleanup
await teardown()
```

In terms of avoiding the `bind` call, yeah we could just move the teardown function into the setup (now that it's not exposed manually) 👍🏻